README.adoc

```diff
@@ -15,18 +22,543 @@ image:https://circleci.com/gh/edn-query-language/eql/tree/master.svg?style=svg["
+                                        :dispatch-key :chat.entry/timestamp
+                                        :key          :chat.entry/timestamp}]}
+                          {:type      :union-entry
+                           :union-key :audio/id
+                           :query     [:audio/id :audio/url :audio/duration :chat.entry/timestamp]
+                           :children  [{:type :prop, :dispatch-key :audio/id, :key :audio/id}
+                                       {:type :prop, :dispatch-key :audio/url, :key :audio/url}
+                                       {:type         :prop
+                                        :dispatch-key :audio/duration
+                                        :key          :audio/duration}
+                                       {:type         :prop
+                                        :dispatch-key :chat.entry/timestamp
+                                        :key          :chat.entry/timestamp}]}
+                          {:type      :union-entry
+                           :union-key :photo/id
+                           :query     [:photo/id
+                                       :photo/url
+                                       :photo/width
+                                       :photo/height
+                                       :chat.entry/timestamp]
+                           :children  [{:type :prop, :dispatch-key :photo/id, :key :photo/id}
+                                       {:type :prop, :dispatch-key :photo/url, :key :photo/url}
+                                       {:type :prop, :dispatch-key :photo/width, :key :photo/width}
+                                       {:type :prop, :dispatch-key :photo/height, :key :photo/height}
+                                       {:type         :prop
+                                        :dispatch-key :chat.entry/timestamp
+                                        :key          :chat.entry/timestamp}]}]}]}]}
+----
+
+=== Mutations
+
+Mutations in EQL are used to represent operation calls, usually to do something
+that will cause a side effect on the world.
+
+A mutation is represented by a list of two elements; the first is the symbol
+that names the mutation and the second is a map with input data.
+
+[source,clojure]
+----
+[(call.some/operation {:data "input"})]
+
+; ast
+
+{:type :root
+ :children
+ [{:dispatch-key call.some/operation
+   :key          call.some/operation
+   :params       {:data "input"}
+   :meta         {:line 610, :column 17}
+   :type         :call}]}
+----
+
+NOTE: Mutations and parameters are very like each other, their main difference
+is that once uses symbols as keys, the other uses one of the read options (properties,
+idents, joins).
+
+Mutations can appear anywhere in the query, root or sub-queries (although if that's supported
+by the parser it's up to each implementation).
+
+==== Mutation Joins
+
+A mutation call can also have a join to represent the data to pull from the
```

Comment:
A mutation may have a return value, and that return value can be a graph; therefore, it makes sense that EQL support the ability to describe what portion of the available returned graph should be returned.  The support for mutation graph return values is done by combining the syntax of a join with the syntax of a mutation: